dynamic/info.go

```diff
@@ -84,7 +84,8 @@ func providerInfo(ctx context.Context, p run.Provider, value parameterize.Value)
 		return prov, err
 	}
 
-	err := prov.ComputeTokens(tokens.SingleModule(p.Name()+"_", "index", tokens.MakeStandard(p.Name())))
+	err := prov.ComputeTokens(tokens.SingleModule(
+		prov.GetResourcePrefix(), "index", tokens.MakeStandard(p.Name())))
```

Comment:
I can see how this fixes this specific error, and I can see that specific case tested in the new test, but I'm not sure I follow why we don't need the "_" for other resources?

Reply:
This is basically an instance of "we were calling `tokens.SingleModule` wrong. It already knows to split on `"_"`.